plugins/out_es/es.h

```diff
@@ -61,6 +61,8 @@ struct flb_elasticsearch {
     /* one for the standard chain provider, one for sts assume role */
     struct flb_tls *aws_sts_tls;
     char *aws_session_name;
+    char *aws_service_name;
+    struct mk_list *aws_unsigned_headers;
```

Comment:
typically in code in FLB that I've seen we have just `struct mk_list` no pointer on context that way we don't have to alloc it and its simpler I think...

Reply:
I had it like that at first, however there is an issue with just having a struct directly nested:

When you run the clean up logic, you need to know whether the slist has been initialized so that you can call:
`flb_slist_destroy(ctx->aws_unsigned_headers);`

 accordingly.

aws_unsigned_headers might not be initialized depending on when the initialization script fails.

Thus using a pointer serves as a double purpose:
1. Let us know if the slist is created
2. store the slist

I could also have made a boolean to track this, but I think it is more straight forward and conforms better with the other code to just allocate memory.

Reply:
I see. I think that's fine but its not the only way to do it. 

There is:

```
static inline int mk_list_is_set(struct mk_list *head)
{
    if (head->next && head->prev) {
        return 0;
    }

    return -1;
}
```

If you just embed the mk_list struct on the context and alloc it with calloc then you can use this function because the next and prev pointers will be initialized to NULL.